src/typechecker/FStar.TypeChecker.Rel.fst

```diff
@@ -3203,6 +3203,11 @@ and solve_t' (env:Env.env) (problem:tprob) (wl:worklist) : solution =
     let try_match_heuristic env orig wl s1 s2 t1t2_opt =
         let try_solve_branch scrutinee p =
             let (Flex (_t, uv, _args), wl) = destruct_flex_t env scrutinee wl  in
+            //
+            // This is dropping the returned guard
+            //   that contains implicits introduced for bvs, dot terms etc.
+            // Why?
```

Comment:
I also recall that we discussed retaining the guard here, so that we don't forget to check that the any implicits introduced here are actually solved. You decided against it?